src/library_fs.js

```diff
@@ -1100,6 +1101,9 @@ mergeInto(LibraryManager.library, {
       if (length < 0 || position < 0) {
         throw new FS.ErrnoError(ERRNO_CODES.EINVAL);
       }
+      if (stream.fd === null) {
```

Comment:
how about adding a helper function, `isClosed(stream)` that would check this? then it would be self-explanatory, otherwise it's not clear what the `fd` being `null` implies.

Aside from that, lgtm with a test. Thanks!